libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt

```diff
@@ -282,32 +292,26 @@ class MapboxTripNotification constructor(
     }
 
     private fun getManeuverResource(step: LegStep): Int {
-        val maneuver = step.maneuver()
-        val maneuverType = maneuver.type()
-        maneuver.let { stepManeuver ->
-            val maneuverModifier = stepManeuver.modifier()
-            if (!TextUtils.isEmpty(maneuverModifier)) {
-                val drivingSide = step.drivingSide()
-                val isLeftSideDriving = isLeftDrivingSideAndRoundaboutOrRotaryOrUturn(
-                    maneuverType,
-                    maneuverModifier,
-                    drivingSide
-                )
-                return when (isLeftSideDriving) {
-                    true -> {
-                        ManeuverResource.obtainManeuverResource(
-                            maneuverType + maneuverModifier + drivingSide
-                        )
-                    }
-                    else -> {
-                        ManeuverResource.obtainManeuverResource(
-                            maneuverType + maneuverModifier
-                        )
-                    }
-                }
+        val stepManeuver = step.maneuver()
+        val maneuverType = stepManeuver.type()
+        val maneuverModifier = stepManeuver.modifier()
+        val isModifierNotEmpty = TextUtils.isEmpty(maneuverModifier).not()
+        val maneuverResourceString = if (isModifierNotEmpty) {
+            val drivingSide = step.drivingSide()
+            val isLeftSideDriving = isLeftDrivingSideAndRoundaboutOrRotaryOrUturn(
+                maneuverType,
+                maneuverModifier,
+                drivingSide
+            )
+            if (isLeftSideDriving) {
+                maneuverType + maneuverModifier + drivingSide
+            } else {
+                maneuverType + maneuverModifier
             }
```

Comment:
Also it would be great if we could put the `when` clause back rather than using if/else